snapcraft/internal/repo/apt_cache.py

```diff
@@ -205,11 +212,17 @@ def get_installed_packages(self) -> Dict[str, str]:
         return installed
 
     def get_packages_marked_for_installation(self) -> List[Tuple[str, str]]:
-        return [
-            (package.name, package.candidate.version)
-            for package in self.cache.get_changes()
-            if package.marked_install
-        ]
+        packages: List[Tuple[str, str]] = []
+        for package in self.cache.get_changes():
+            if not package.marked_install:
+                continue
+
+            if package.candidate is None:
+                raise errors.PackageNotFoundError(package.name)
+
+            packages.append((package.name, package.candidate.version))
```

Comment:
Would this be a bit more descriptive?
```suggestion
        changed_packages = self.cache.get_changes()
        marked_install_packages = [p for p in changed_packages if p.marked_install]
        
        # TODO add PackagesNotFoundError (plural).
        packages_missing_installation_candidate = [p for p in marked_install_packages if p.candidate is None]
        if len(packages_missing_installation_candidate) > 0:
            raise errors.PackageNotFoundError(packages_missing_installation_candidate[0].name)
        
        return [(p.name, p.candidate.version) for p in marked_install_packages]
```

Reply:
Done.